lib/rules/computed-property-getters.js

```diff
@@ -94,9 +95,25 @@ module.exports = {
       }
     };
 
+    let importedEmberName;
+    let importedComputedName;
+
     return {
+      ImportDeclaration(node) {
```

Comment:
This logic is getting repeated a lot of times. It could be nice to make a more generic `imports` object, and do something like `imports = updateImports(imports, node)`. Then `isComputedProp` could take that `imports` object as an argument.

Reply:
Ya, or even something like:

```js
let imports = new Map();

return {
  ImportDeclaration: gatherImports(imports),

  // ...snip...
}
```

Where the `gatherImports` method returns a function that works as a visitor handler itself, _or_ if you also have custom import logic you need you could do this (with the same API):

```js
let imports = new Map();
let importGatherer = gatherImports(imports);

return {
  ImportDeclaration(node) {
    importGatherer(node);
    // other logic
  }
};
```

Reply:
Definitely agree with this. I may do this as a follow-up so I can focus on this improvement by itself.

Reply:
Working on the new API in this PR: https://github.com/ember-cli/eslint-plugin-ember/pull/910